Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,288 @@
+
+        //create a bounding box so the prop can be scaled proportionally
+        GameObject boundingScaleBox = GameObject.CreatePrimitive(PrimitiveType.Cube);
+        boundingScaleBox.GetComponent<MeshRenderer>().enabled = false;
+        boundingScaleBox.GetComponent<BoxCollider>().enabled = false;
+        boundingScaleBox.transform.position = propTile.transform.position;
+
+        Vector3 menuPropDimension = Vector3.zero;
+
+        //if the model has children, calculate the total bounds of all the children
+        if (menuProp.transform.childCount > 0)
+        {
+            Bounds bounds = new Bounds(menuPropDisplay.transform.position, Vector3.zero);
+            foreach (Renderer renderer in menuProp.GetComponentsInChildren<Renderer>())
+            {
+                bounds.Encapsulate(renderer.bounds);
+            }
+            menuPropDimension = bounds.size;
+        }
+        //only calculate the bounds of the prop
+        else
+        {
+            menuPropDimension = menuPropDisplay.GetComponent<Renderer>().bounds.size;
+        }
+
+        //compare the sides of the menu prop to find the largest side
+        //set the bounding scale box to fit around the menu prop
+        if (menuPropDimension.x >= menuPropDimension.y && menuPropDimension.x >= menuPropDimension.z)
+        {
+            biggestPropSide = menuPropDimension.x;
+            boundingScaleBox.transform.localScale = new Vector3(biggestPropSide, biggestPropSide, biggestPropSide);
+        }
+
+        if (menuPropDimension.y >= menuPropDimension.x && menuPropDimension.y >= menuPropDimension.z)
+        {
+            biggestPropSide = menuPropDimension.y;
+            boundingScaleBox.transform.localScale = new Vector3(biggestPropSide, biggestPropSide, biggestPropSide);
+        }
+
+        if (menuPropDimension.z >= menuPropDimension.x && menuPropDimension.z >= menuPropDimension.y)
+        {
+            biggestPropSide = menuPropDimension.z;
+            boundingScaleBox.transform.localScale = new Vector3(biggestPropSide, biggestPropSide, biggestPropSide);
+        }
+
+        menuPropDisplay.transform.parent = boundingScaleBox.transform;
+        menuPropDisplay.transform.localPosition = new Vector3(0, 0, 0);
+
+        boundingScaleBox.transform.parent = propTile.transform;
+        boundingScaleBox.transform.localScale = new Vector3(.6f, .6f, .6f);
+
+        if (menuPropDisplay.GetComponent<Collider>())
+        {
+            menuPropDisplay.GetComponent<Collider>().enabled = false;
+        }
+
+        menuPropDisplay.AddComponent<slowRotation>();
+        menuPropDisplay.SetActive(true);
+    }
+
+    void triggerClicked(object sender, ClickedEventArgs e)
+    {
+        Ray ray = new Ray(inputManager.hand2.gameObject.transform.position, inputManager.hand2.gameObject.transform.forward);
+        RaycastHit hit;
+        if (Physics.Raycast(ray, out hit, 100))
+        {
+
+            if (hit.collider.transform.parent == propTiles.transform)
```

Comment:
what's wrong with making the propTiles iHittable and running this method through them? > that way its consistent as to how we are handling menu items